satellite/console/service.go

```diff
@@ -141,6 +119,37 @@ func (s *Service) CreateUser(ctx context.Context, user CreateUser, tokenSecret R
 		return nil, errs.New(internalErrMsg)
 	}
 
+	// TODO: remove after vanguard release
+	var registrationToken *RegistrationToken
+	if user.PartnerID != "" {
+		registrationToken, err = s.store.RegistrationTokens().Create(ctx, 1)
+		if err != nil {
+			return nil, errs.New(internalErrMsg)
+		}
+	} else {
+		registrationToken, err = s.store.RegistrationTokens().GetBySecret(ctx, tokenSecret)
+		if err != nil {
+			return nil, errs.New(vanguardRegTokenErrMsg)
+		}
+		if registrationToken.OwnerID != nil {
```

Comment:
same about comments. The nest logic is hard to reason about so maybe some documentation will help